libraries/AP_Param_Helper/AP_Param_Helper.cpp

```diff
@@ -5,9 +5,11 @@
 #if defined(HAL_NEEDS_PARAM_HELPER)
 
 const AP_Param::GroupInfo AP_Param_Helper::var_info[] = {
+    F4LIGHT_HAL_VARINFO
```

Comment:
I know that no other HAL uses this now, but, as it is a generic library, we should put a macro around this:

```
#if CONFIG_HAL_BOARD == HAL_BOARD_F4LIGHT
    F4LIGHT_HAL_VARINFO
#endif
```

Reply:
this is not good approach, it should be selected by feature, not by name. Added #if  defined() so any board could to use parameters this way. As it did with HAL_REQUIRES_PARAM_HELPER

Reply:
That's fine.